docs/tutorial.md

```diff
@@ -0,0 +1,526 @@
+    private fun EmailItem(email: Email, modifier: Modifier = Modifier, onClick: () -> Unit) {
+      // ...
+    }
+    ```
+
+Finally, let's handle this event in our presenter.
+
+=== "InboxPresenter"
+    ```kotlin
+    class InboxPresenter(
+      private val navigator: Navigator,
+      private val emailRepository: EmailRepository
+    ) : Presenter<InboxScreen.State> {
+      @Composable
+      override fun present(): InboxScreen.State {
+        val emails = emailRepository.getEmails()
+        return InboxScreen.State(emails) { event ->
+          when (event) {
+            // Navigate to the detail screen when an email is clicked
+            is EmailClicked -> navigator.goTo(DetailScreen(event.emailId))
+          }
+        }
+      }
+    }
+    ```
+
+This demonstrates how we can navigate forward in our app and pass data with it. Let's see how we can navigate back.
+
+## Navigating back
+
+Naturally, navigation can't be just one way. The opposite of `Navigator.goTo()` is `Navigator.pop()`, which pops the back stack back to the previous screen. To use this, let's add a back button to our detail screen and wire it up to a `Navigator`.
+
+=== "DetailScreen"
+    ```kotlin
+    data class DetailScreen(val emailId: String) : Screen {
+      data class State(
+        val email: Email,
+        val eventSink: (Event) -> Unit
+      ) : CircuitUiState
+      sealed class Event : CircuitUiEvent {
+        data object BackClicked : Event()
+      }
+    }
+    ```
+
+=== "DetailContent"
+    ```kotlin
+    @Composable
+    fun EmailDetail(state: DetailScreen.State, modifier: Modifier = Modifier) {
+      Column(modifier = modifier) {
+        IconButton(onClick = { state.eventSink(DetailScreen.Event.BackClicked) }) {
+          Icon(Icons.Default.ArrowBack, contentDescription = "Back")
+        }
+        // Remaining detail UI...
+      }
+    }
+    ```
+
+=== "DetailPresenter"
+    ```kotlin
+    class DetailPresenter(
+      private val screen: DetailScreen,
+      private val navigator: Navigator,
+      private val emailRepository: EmailRepository,
+    ) : Presenter<DetailScreen.State> {
+      @Composable
+      override fun present(): DetailScreen.State {
+        val email = emailRepository.getEmail(screen.emailId)
+        return DetailScreen.State(email) { event ->
+          when (event) {
+            DetailScreen.Event.BackClicked -> navigator.pop()
+          }
+        }
+      }
+      // ...
+    }
+    ```
+
+On Android, `NavigableCircuitContent` automatically hooks into `BackHandler` to automatically pop on system back presses. On Desktop, it's recommended to wire the ESC key.
+
+## Conclusion
+
+This is just a brief introduction to Circuit. For more information see various docs on the site, samples in the repo, the [API reference](../api/0.x/index.html), and check out other Circuit tools like circuit-retained, CircuitX, factory code gen, overlays, testing, multiplatform, and more.
```

Comment:
Add links to all of the different tools.

Reply:
[7517fe6](https://github.com/slackhq/circuit/pull/1158/commits/7517fe63109e3a657c5fdf184cf10bbe1a59730d)

[44f7b9f](https://github.com/slackhq/circuit/pull/1158/commits/44f7b9fe83f651a807cf0e0bf76fd0d722a57985)